File: swift/BCURUI/Sources/BCURUI/URScanFeedbackConfig.swift
import Foundation

public struct URScanFeedbackConfig: Sendable {
    public var hapticEnabled: Bool
    public var soundEnabled: Bool
    public var clickSoundURL: URL?
    public var successSoundURL: URL?
    public var failureSoundURL: URL?

    public init(
        hapticEnabled: Bool = true,
        soundEnabled: Bool = false,
        clickSoundURL: URL? = nil,
        successSoundURL: URL? = nil,
        failureSoundURL: URL? = nil
    ) {
        self.hapticEnabled = hapticEnabled
        self.soundEnabled = soundEnabled
        self.clickSoundURL = clickSoundURL
        self.successSoundURL = successSoundURL
        self.failureSoundURL = failureSoundURL
    }

    public static let `default` = URScanFeedbackConfig()

    public static func hapticOnly(_ enabled: Bool) -> URScanFeedbackConfig {
        URScanFeedbackConfig(hapticEnabled: enabled)
    }
}

File: kotlin/bc-urui/src/main/kotlin/com/blockchaincommons/bcurui/URScanFeedbackConfig.kt
package com.blockchaincommons.bcurui

data class URScanFeedbackConfig(
    val hapticEnabled: Boolean = true,
    val soundEnabled: Boolean = false,
    val clickSoundResId: Int? = null,
    val successSoundResId: Int? = null,
    val failureSoundResId: Int? = null,
) {
    companion object {
        val DEFAULT = URScanFeedbackConfig()
        fun hapticOnly(enabled: Boolean) = URScanFeedbackConfig(hapticEnabled = enabled)
    }
}
